src/Razor/src/Microsoft.VisualStudioCode.RazorExtension/Services/RazorWorkspaceService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.VisualStudioCode.RazorExtension.Services;
+
+[ExportRazorStatelessLspService(typeof(RazorWorkspaceService)), Shared]
+[method: ImportingConstructor]
+file class WorkspaceService(ILoggerFactory loggerFactory) : RazorWorkspaceService
+{
+    private readonly ILoggerFactory _loggerFactory = loggerFactory;
+    private readonly ILogger _logger = loggerFactory.CreateLogger<RazorWorkspaceService>();
```

Comment:
This should probably create a logged for `WorkspaceService`, though being a file class, that type name probably looks ugly.